### DISCUSSION TITLE
How do I customize the Fixture used by the XUnit AutoDataAttribute?

### DISCUSSION THREAD

**Discussion by HurricanKai** [AutoFixture/AutoFixture#1191] [Q&A]
See Title, I'm using AutoFixture to simply generate numbers, and it's working well, but I have a custom number type `Half` that I want to essentially be the same as float just for 16 bytes.
I can't find anything on how to customize the AutoDataAttribute Fixture though.

**Comment by Kralizek** [comment]
What you should do in this case is to create a class inheriting from `AutoDataAttribute` and use the constructor that accepts a `Func<IFixture>`.

Here is an example

```csharp
public class MyAutoDataAttribute : AutoDataAttribute
{
    public MyAutoDataAttribute() : base(CreateFixture) { }

    private static IFixture CreateFixture()
    {
        var fixture = new Fixture();

        // your customizations here

        return fixture
    }
}
```

Then you proceed by decorating your tests with this new attribute.

[Here](https://github.com/emgdev/dotnet-utils/blob/master/tests/Tests.ServiceModel/CustomAutoDataAttribute.cs) I am using the equivalent pattern for NUnit 3, hopefully it helps :)
